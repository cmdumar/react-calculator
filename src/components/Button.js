import propTypes from 'prop-types';

function Button({
  name, clickHandler, color, wide,
}) {
  return (
    <button className={`${color ? 'orange' : null} ${wide ? 'wide' : null}`} type="button" onClick={() => clickHandler(name)}>
      {name}
    </button>
  );
}

Button.propTypes = {
  name: propTypes.string.isRequired,
  clickHandler: propTypes.func.isRequired,
  color: propTypes.bool,
  wide: propTypes.bool,
};

Button.defaultProps = {
  color: false,
  wide: false,
};

export default Button;
