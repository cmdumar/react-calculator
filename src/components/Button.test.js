import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

test('Renders button on the page', () => {
  const btn = renderer.create(<Button name="23" clickHandler={() => 'something'} />);

  expect(btn).toMatchSnapshot();
});
