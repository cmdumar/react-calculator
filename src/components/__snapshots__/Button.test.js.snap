// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders button on the page 1`] = `
<button
  className="null null"
  onClick={[Function]}
  type="button"
>
  23
</button>
`;
